app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/RoleActivity.java

```diff
@@ -101,39 +101,28 @@ public void onSaveInstanceState(Bundle savedInstanceState) {
     }
 
     /**
-     * Gets the View of the Gunner button.
+     * Updates the view of the roles in the RoleActivity so that no more than 1 player may choose
+     * the same role.
      *
-     * @return a View of the button.
+     * @param message the message received.
      */
-    public static View getGunnerView() {
-        return gunnerView;
-    }
-
-    /**
-     * Gets the View of the Engineer button.
-     *
-     * @return a View of the button.
-     */
-    public static View getEngineerView() {
-        return engineerView;
-    }
-
-    /**
-     * Gets the View of the Scientist button.
-     *
-     * @return a View of the button.
-     */
-    public static View getScientistView() {
-        return scientistView;
-    }
-
-    /**
-     * Gets the View of the Janitor button.
-     *
-     * @return a View of the button.
-     */
-    public static View getJanitorView() {
-        return janitorView;
+    public static void updateRoleAvailability(RoleChosenMessage message) {
+        switch(message.getRole()){
+            case GUNNER:
+                gunnerView.setEnabled(false);
+                break;
+            case ENGINEER:
+                engineerView.setEnabled(false);
+                break;
+            case JANITOR:
+                janitorView.setEnabled(false);
+                break;
+            case SCIENTIST:
+                scientistView.setEnabled(false);
+                break;
+            default:
+                break;
+        }
```

Comment:
A shame this was implemented using a switch statement. Will refactor while implementing and integrating server-side validation of role allocation.